1.19.0-alpha.1

matteobruni opened the pull request:
feat: spin movement is now working, still needs improvements
refactor: presets and shapes now can be imported and used in all components, more docs will be added
feat: added responsive options
feat: distance property in particles.move is now an object with horizontal and vertical properties
fix: fixed absorber plugin
chore: improved black holes (absorbers with orbits)
feat: particles updater and interactors are now part of the plugin system
feat: destroy section in particles options, this can configure particle split on destroy

matteobruni committed:
feat: spin movement is now working, still needs improvements

ghost commented:
## DeepCode's analysis on [#abad52](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/3c7da3e289543ad510b0f6e3f448bd008f1292d1/matteobruni/tsparticles/abad525077eefd451129836b2ab9babfe3289a4b/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=5&) found:

- :information_source: **5** minor issues. :point_down:
- :heavy_check_mark: **2** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/main.slim.ts#L98">main.slim.ts:98</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/main.slim.ts#L111">main.slim.ts:111</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/main.slim.ts#L125">main.slim.ts:125</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/3c7da3e289543ad510b0f6e3f448bd008f1292d1/matteobruni/tsparticles/abad525077eefd451129836b2ab9babfe3289a4b/pr/_/%2Fcore%2Fmain%2Fsrc%2Fmain.slim.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Switch statement should include a 'default' case Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/Updaters/OpacityUpdater.ts#L37">OpacityUpdater.ts:37</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/Core/Particle/InteractionManager.ts#L23">InteractionManager.ts:23</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/Updaters/SizeUpdater.ts#L33">SizeUpdater.ts:33</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/3c7da3e289543ad510b0f6e3f448bd008f1292d1/matteobruni/tsparticles/abad525077eefd451129836b2ab9babfe3289a4b/pr/_/%2Fcore%2Fmain%2Fsrc%2FUpdaters%2FOpacityUpdater.ts/javascript%2Ftslint%2Fswitch-default/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/ShapeDrawers/ImageDrawer.ts#L63">ImageDrawer.ts:63</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/Utils/Plugins.ts#L20">Plugins.ts:20</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/abad525077eefd451129836b2ab9babfe3289a4b/core/main/src/Utils/Plugins.ts#L22">Plugins.ts:22</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/3c7da3e289543ad510b0f6e3f448bd008f1292d1/matteobruni/tsparticles/abad525077eefd451129836b2ab9babfe3289a4b/pr/_/%2Fcore%2Fmain%2Fsrc%2FShapeDrawers%2FImageDrawer.ts/javascript%2Ftslint%2Farray-type/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/3c7da3e289543ad510b0f6e3f448bd008f1292d1/matteobruni/tsparticles/abad525077eefd451129836b2ab9babfe3289a4b/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=5&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
refactor: presets now can be imported and used in all components, more docs will be added

matteobruni committed:
build: shapes are converted like presets

matteobruni committed:
build: fixed deep scan issues

matteobruni committed:
fix(angular): disabled Ivy as suggested

matteobruni committed:
build: updated packages

dependabot[bot] committed:
build(deps): bump @angular/animations from 10.2.2 to 10.2.3

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 10.2.2 to 10.2.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.2.3/packages/animations)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 10.2.2 to 10.2.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.2.3/packages/platform-browser-dynamic)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1009 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-10.2.3

build(deps): bump @angular/platform-browser-dynamic from 10.2.2 to 10.2.3

matteobruni committed:
Merge pull request #1008 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-10.2.3

build(deps): bump @angular/animations from 10.2.2 to 10.2.3

matteobruni committed:
docs: preparing readme for big circles, WIP

matteobruni committed:
build: added stats.ts to main demo

dependabot[bot] committed:
build(deps-dev): bump sass-loader from 10.0.5 to 10.1.0

Bumps [sass-loader](https://github.com/webpack-contrib/sass-loader) from 10.0.5 to 10.1.0.
- [Release notes](https://github.com/webpack-contrib/sass-loader/releases)
- [Changelog](https://github.com/webpack-contrib/sass-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/sass-loader/compare/v10.0.5...v10.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: added stats.ts to jquery demo, various updates

matteobruni committed:
Merge pull request #1011 from matteobruni/dependabot/npm_and_yarn/dev/sass-loader-10.1.0

build(deps-dev): bump sass-loader from 10.0.5 to 10.1.0

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
Update package.json

matteobruni committed:
build: fixed vue3 demo
feat: changed vue3 component with particlesInit and particlesLoaded methods, more docs coming

matteobruni committed:
build: disabled vue3 demo build

dependabot[bot] committed:
build(deps): bump preact from 10.5.5 to 10.5.7

Bumps [preact](https://github.com/preactjs/preact) from 10.5.5 to 10.5.7.
- [Release notes](https://github.com/preactjs/preact/releases)
- [Commits](https://github.com/preactjs/preact/compare/10.5.5...10.5.7)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1015 from matteobruni/dependabot/npm_and_yarn/dev/preact-10.5.7

build(deps): bump preact from 10.5.5 to 10.5.7

matteobruni committed:
build: removed yarn workspaces for issues with dependencies resolution

matteobruni committed:
build: updated angular packages in demo project

matteobruni committed:
build: fixed vue 2.x demo, needs a fix

matteobruni committed:
build: restored some vue 2 demo code, the workaround is still there

matteobruni committed:
build: updated docs for vue components

matteobruni committed:
build: added commented not working code in vue demo

matteobruni committed:
feat: added responsive options, closes #1004

matteobruni committed:
feat: distance property in particles.move is now an object with horizontal and vertical properties

matteobruni committed:
build: updated vue 2.x plugin syntax

matteobruni committed:
fix(tests): fixed tests

matteobruni committed:
fix: fixed absorber plugin

matteobruni committed:
chore: improved black holes (absorbers with orbits)

matteobruni committed:
fix: fixed vue plugin, thanks to @tony19

matteobruni committed:
build: fixed warning for DeepCode

matteobruni committed:
build: updated dependencies

matteobruni committed:
build: updated lockfiles

matteobruni committed:
build: restored istanbul for coverage

matteobruni committed:
feat: improved particle spin behavior

matteobruni committed:
fix: fixed manual particles with groups
refactor: improved some promises

matteobruni committed:
build: code prettified

matteobruni committed:
build: updated stats.ts

matteobruni committed:
build: updated demo.js for main project

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.7.0 to 4.8.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.7.0 to 4.8.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.8.0/packages/parser)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.7.0 to 4.8.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.8.0/packages/eslint-plugin)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1022 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.8.0

build(deps-dev): bump @typescript-eslint/parser from 4.7.0 to 4.8.0

matteobruni committed:
Merge pull request #1023 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.8.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.7.0 to 4.8.0

matteobruni committed:
build: updated lockfile

matteobruni committed:
Update bug_report.md

matteobruni committed:
Update custom.md

matteobruni committed:
Update feature_request.md

matteobruni committed:
Merge pull request #1025 from matteobruni/issue-templates

Update Issue Templates

matteobruni committed:
feat: particles updater and interactors are now part of the plugin system

matteobruni committed:
refactor: splitted some complex methods in Linker interactor

matteobruni committed:
refactor: migrated to const enums to have a smaller bundle size

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
feat: destroy section in particles options, this can configure particle split on destroy fixes #607

matteobruni committed:
build: updated deps

matteobruni committed:
chore: improved particle split

matteobruni committed:
chore: improved particle split

matteobruni committed:
chore(release): published new version

 - ng-particles@2.2.0-alpha.1
 - inferno-particles@1.2.0-alpha.1
 - jquery-particles@1.19.0-alpha.1
 - preact-particles@1.19.0-alpha.1
 - react-tsparticles@1.19.0-alpha.1
 - svelte-particles@1.2.0-alpha.1
 - particles.vue@2.2.0-alpha.1
 - particles.vue3@1.2.0-alpha.1
 - tsparticles-editor@1.19.0-alpha.1
 - tsparticles@1.19.0-alpha.1
 - ng-particles-demo@1.4.0-alpha.1
 - editor-demo@1.4.0-alpha.1
 - inferno-particles-demo@1.2.0-alpha.1
 - jquery-particles-demo@1.4.0-alpha.1
 - tsparticles-demo@1.4.0-alpha.1
 - preact-particles-demo@1.4.0-alpha.1
 - react-tsparticles-demo@1.4.0-alpha.1
 - svelte-demo@1.2.0-alpha.1
 - particles.vue-demo@1.4.0-alpha.1
 - particles.vue3-demo@1.2.0-alpha.1
 - tsparticles-preset-big-circles@1.4.0-alpha.1
 - tsparticles-preset-fire@1.4.0-alpha.1
 - tsparticles-preset-fountain@1.4.0-alpha.1
 - tsparticles-preset-links@1.4.0-alpha.1
 - tsparticles-preset-sea-anemone@1.4.0-alpha.1
 - tsparticles-preset-snow@1.4.0-alpha.1
 - tsparticles-preset-stars@1.4.0-alpha.1
 - tsparticles-shape-bubble@1.4.0-alpha.1
 - tsparticles-shape-heart@1.4.0-alpha.1
 - tsparticles-shape-multiline-text@1.4.0-alpha.1
 - tsparticles-shape-spiral@1.4.0-alpha.1
 - cra-template-particles-typescript@1.2.0-alpha.1
 - cra-template-particles@1.2.0-alpha.1
